File: app/src/main/java/fr/xgouchet/chronorg/inject/modules/FormatterModule.java
package fr.xgouchet.chronorg.inject.modules;

import org.joda.time.Duration;
import org.joda.time.ReadableInstant;

import dagger.Module;
import dagger.Provides;
import fr.xgouchet.chronorg.data.formatters.DurationFormatter;
import fr.xgouchet.chronorg.data.formatters.Formatter;
import fr.xgouchet.chronorg.data.formatters.ReadableInstantFormatter;
import fr.xgouchet.chronorg.inject.annotations.ApplicationScope;

/**
 * @author Xavier Gouchet
 */
@Module
public class FormatterModule {

    @Provides
    @ApplicationScope
    public Formatter<ReadableInstant> provideReadableInstantFormatter() {
        return new ReadableInstantFormatter();
    }

    @Provides
    @ApplicationScope
    public Formatter<Duration> provideDurationFormatter() {
        return new DurationFormatter();
    }

}

File: app/src/main/java/fr/xgouchet/chronorg/di/FormatterModule.kt
package fr.xgouchet.chronorg.di

import fr.xgouchet.chronorg.ui.formatter.Formatter
import fr.xgouchet.chronorg.ui.formatter.InstantFormatter
import fr.xgouchet.chronorg.ui.formatter.IntervalFormatter
import org.joda.time.Instant
import org.joda.time.Interval
import org.kodein.di.Kodein
import org.kodein.di.generic.bind
import org.kodein.di.generic.provider

val FormatterModule = Kodein.Module(name = "Formatter") {
    bind<Formatter<Instant>>() with provider { InstantFormatter() }
    bind<Formatter<Interval>>() with provider { IntervalFormatter() }
}
